transforms/universal/blocklist/kfp_ray/blocklist_wf.py

```diff
@@ -0,0 +1,231 @@
+        wait_cluster_up_tmout - time to wait for cluster up, sec
+        wait_job_ready_tmout - time to wait for job ready, sec
+        wait_print_tmout - time between prints, sec
+        http_retries - http retries for API server calls
+    :param data_s3_access_secret - s3 access secret
+    :param data_s3_config - s3 configuration
+    :param data_max_files - max files to process
+    :param data_num_samples - num samples to process
+    :param runtime_actor_options - actor options
+    :param runtime_pipeline_id - pipeline id
+    :param runtime_code_location - code location
+    :param blocked_domain_list_path - S3 url to blocked domain lists
+    :param annotation_column_name - annotation column
+    :param source_url_column_name - source document url column
+    :return: None
+    """
+    # create clean_up task
+    clean_up_task = cleanup_ray_op(ray_name=ray_name, run_id=run_id, server_url=server_url)
+    ComponentUtils.add_settings_to_component(clean_up_task, 60)
+    # pipeline definition
+    with dsl.ExitHandler(clean_up_task):
+        # compute execution params
+        compute_exec_params = compute_exec_params_op(
+            worker_options=ray_worker_options,
+            actor_options=runtime_actor_options,
+            data_s3_config=data_s3_config,
+            data_max_files=data_max_files,
+            data_num_samples=data_num_samples,
+            data_checkpointing=data_checkpointing,
+            data_data_sets=data_data_sets,
+            data_files_to_use=data_files_to_use,
+            runtime_pipeline_id=runtime_pipeline_id,
+            runtime_job_id=run_id,
+            runtime_code_location=runtime_code_location,
+            blocklist_blocked_domain_list_path=blocklist_blocked_domain_list_path,
+            blocklist_annotation_column_name=blocklist_annotation_column_name,
+            blocklist_source_url_column_name=blocklist_source_url_column_name,
+        )
+        ComponentUtils.add_settings_to_component(compute_exec_params, ONE_HOUR_SEC * 2)
+        # start Ray cluster
+        ray_cluster = create_ray_op(
+            ray_name=ray_name,
+            run_id=run_id,
+            ray_head_options=ray_head_options,
+            ray_worker_options=ray_worker_options,
+            server_url=server_url,
+            additional_params=additional_params,
+        )
+        ComponentUtils.add_settings_to_component(ray_cluster, ONE_HOUR_SEC * 2)
+        ray_cluster.after(compute_exec_params)
+        # Execute job
+        execute_job = execute_ray_jobs_op(
+            ray_name=ray_name,
+            run_id=run_id,
+            additional_params=additional_params,
+            exec_params=compute_exec_params.output,
+            exec_script_name=EXEC_SCRIPT_NAME,
+            server_url=server_url,
+        )
+
+        ComponentUtils.add_settings_to_component(execute_job, ONE_WEEK_SEC)
+        ComponentUtils.set_s3_env_vars_to_component(execute_job, data_s3_access_secret)
```

Comment:
This is only applicable for KFPv1. Suggest replacing with the following block:
```suggestion
        if os.getenv("KFPv2", "0") == "1":
            from kfp import kubernetes
            
            # FIXME: Due to kubeflow/pipelines#10914, secret names cannot be provided as pipeline arguments.
            # As a workaround, the secret name is hard coded.
            env2key = ComponentUtils.set_secret_key_to_env(prefix="jjj")
            kubernetes.use_secret_as_env(task=execute_job, secret_name=S3_SECRET, secret_key_to_env=env2key)
        else:
            ComponentUtils.set_s3_env_vars_to_component(execute_job, data_s3_access_secret)
```